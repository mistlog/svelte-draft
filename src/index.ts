export * from "./transcriber/svelte-transcriber";

export * from "./section/script-section";
export * from "./section/import-section";
export * from "./section/template-section";