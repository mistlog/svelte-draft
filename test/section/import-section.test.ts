import { SvelteTranscriber } from "../../src";

describe("translate import section", () =>
{
    test("extrat import from code", () =>
    {
        const code = `
            import Nested from "./Nested.svelte.tsx";
            import { count as $count } from "./store.js.tsx";

            export default function App()
            {
                let name = "";
                console.log($count);
                
                <p>This is a paragraph.</p>;
                <Nested/>;
            }
        `;

        const { import_section } = new SvelteTranscriber(code).TranscribeToSections();
        expect(import_section).toMatchSnapshot();
    })

    test("remove ts related import", () =>
    {
        const code = `
            import {ITest, Test} from "./Nested.svelte.tsx";

            export default function App()
            {
                const test1: ITest = {};
                const test2 = new Test();
            }
        `;

        const { import_section } = new SvelteTranscriber(code).TranscribeToSections();
        expect(import_section).toMatchSnapshot();
    })
})