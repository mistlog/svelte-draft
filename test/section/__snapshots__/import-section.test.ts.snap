// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translate import section extrat import from code 1`] = `
"import Nested from \\"./Nested.svelte\\";
import { count as $count } from \\"./store.js\\";"
`;

exports[`translate import section remove ts related import 1`] = `"import { Test } from \\"./Nested.svelte\\";"`;
